examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1262 @@
+    // the solution and of the level set.
+    parallel::distributed::Triangulation<dim, dim> triangulation;
+    ConditionalOStream                             pcout;
+    TimerOutput                                    computing_timer;
+
+    // However, the degrees of their FE spaces may be different.
+    const unsigned int fe_degree;
+    const unsigned int level_set_fe_degree;
+
+    // The first bulk quadrature is required for the
+    // for TraceFEM stabilization, while the integration over implicit surface
+    // is based on the last, one-dimensional rule.
+    const QGauss<dim> cell_quadrature;
+    const QGauss<1>   quadrature_1D;
+
+    // We need two separate FE spaces.
+    // The first manages the TraceFEM space which is active on intersected
+    // element. The second manages the discrete
+    // level set function that describes the geometry of the surface.
+    hp::FECollection<dim> fe_collection;
+    const FE_Q<dim>       level_set_fe;
+
+    // The corresponding DoFHandler objects are given by
+    DoFHandler<dim> dof_handler;
+    DoFHandler<dim> level_set_dof_handler;
+
+    // Since we will adaptively refine the bulk triangulation, two constraints
+    // are needed: one for the solution space and another for the level set
+    // space.
+    AffineConstraints<double> constraints;
+    AffineConstraints<double> level_set_constraints;
+
+    // Discrete vectors initialized with dof_handler and level_set_dof_handler.
+    VectorType    completely_distributed_solution;
+    VectorType    locally_relevant_solution;
+    VectorType    locally_relevant_exact;
+    VectorType    level_set;
+    Vector<float> activeFE_indicator;
+
+    // Mesh_classifier separates intersected elements and non-intersected ones
+    // in the fe_collection
+    NonMatching::MeshClassifier<dim> mesh_classifier;
+    const MappingQ1<dim>             mapping;
+
+    // Any TraceFEM need a stabilization, and we choose the normal-gradient,
+    // volume stabilization.
+    NormalGradientVolumeStabilization<dim> stabilization_scheme;
+
+    // Discrete right-hand side and the final matrix correspondent to
+    // dof_handler.
+    VectorType      global_rhs;
+    MatrixType      global_matrix;
+    SparsityPattern sparsity_pattern;
+    IndexSet        locally_owned_dofs;
+    IndexSet        locally_relevant_dofs;
+
+    // Depending on the type of the quadrature, surface, face or volume, we need
+    // to define different update flags.
+    NonMatching::RegionUpdateFlags surface_update_flags;
+    UpdateFlags                    cell_update_flags;
+
+    // The following variables are used to display the results of the
+    // convergence test:
+    ConvergenceTable convergence_table;
+    unsigned int     number_of_iterations;
+    double           average;
+    double           area;
+    double           error_L2_sqr;
+    double           error_H1_sqr;
+    double           error_stab_sqr;
```

Comment:
Yes, I think the right place to deal with this is to compute it in one place and immediate put it into the table, rather than spread all of this out over two functions and one class member variable. It is far easier to understand things that are local, rather than logic spread over multiple functions.